speed-dreams/source-2.2.3/src/modules/userinterface/legacymenu/mainscreens/EndExperimentMenu.cpp

```diff
@@ -0,0 +1,49 @@
+#include <tgfclient.h>
+
+#include "legacymenu.h"
+
+#include "Mediator.h"
+#include "EndExperimentMenu.h"
+#include "mainmenu.h"
+
+static void* s_menuHandle = NULL;
+
+static void OnAcceptExit(void* /* dummy */)
+{
+    LmRaceEngine().abortRace();  // Do cleanup to get back correct setup files
+    LegacyMenu::self().quit();
+}
+
+void* EndExperimentInit(void* p_menuHandle, RaceEndType p_saveWayVersion)
+{
+    if (s_menuHandle)
+    {
+        GfuiScreenRelease(s_menuHandle);
+    }
```

Comment:
All other menu Init-functions start with this:
```suggestion
    // Initialize only once.
    if (MenuHandle)
        return MenuHandle;
```
Is that not better?

Reply:
De ExitMenuInit functie in exitmenu.cpp doet dat niet zo; daar heb ik dit vandaan.